bpf/nodejs.c

```diff
@@ -0,0 +1,91 @@
+#include "vmlinux.h"
+#include "bpf_helpers.h"
+#include "bpf_dbg.h"
+#include "pid.h"
+#include "ringbuf.h"
+#include "nodejs.h"
+
+char __license[] SEC("license") = "Dual MIT/GPL";
+
+volatile const s32 async_wrap_async_id_off = 0;
+volatile const s32 async_wrap_trigger_async_id_off = 0;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_LRU_HASH);
+    __type(key, u64);   // the pid_tid 
+    __type(value, u64); // the last AsyncWrap *
+    __uint(max_entries, 1000); // 1000 nodejs services, small number, nodejs is single threaded
+    __uint(pinning, LIBBPF_PIN_BY_NAME);
+} async_reset_args SEC(".maps");
+
+SEC("uprobe/node:AsyncReset")
```

Comment:
Tracks this NodeJS function to remember the AsyncWrap NodeJS pointer. We then use the pointer to read the async_id_ and async_trigger_id_ in EmitAsyncInit.